### DISCUSSION TITLE
Feature Request: Account Community Center Upgrades (while hiding gems spent)

### DISCUSSION THREAD

**Discussion by ptlthg** [HypixelDev/PublicAPI#647] [Ideas]
It's been requested before, but I am assuming hasn't been a priority to include because Hypixel understandably doesn't want to expose any monetization data. What I am proposing is a way to expose the free unlockable community center upgrades without giving any possible way for someone to deduce if gems were spent or not. **I am only proposing a way to display FREE upgrades, not those that cost gems to start.**

Profile upgrades are already in the API while also being able to know how many gems were used though (which is a bit confusing). If this is just a problem on where to expose this data, putting it in the `/player` endpoint could work for now.

Here is an example of how this could work, imagine we have a community center upgrade which takes 24 hours to unlock. After starting the upgrades, the following rules are applied:

**Tier 1:** 1 day - Visible in API 3 days after user unlocks it (the duration times 3)
**Tier 2:** 2 days - Visible in API 6 days after tier 1 is visible in the API
**Tier 3:** 3 days - Visible in API 9 days after tier 2 is visible in the API

This type of system would mean that consumers of the API would have no possible way of knowing if gems were used to skip the cooldowns. In the case of someone spending gems to unlock multiple upgrades before a cycle above could complete, the next upgrades should be queued to be visible after the first. (Note: timestamps on upgrades can either be left off or the delayed timestamps can be used)

### Why do we want this?
Account upgrades that provide players with stats are impossible to detect right now (outside of mods). This is only really important for Magic Find and Farming Fortune, but it hurts the user experience to require players to enter these stats manually.

Even if upgrade states are delayed by months, it would still be greatly appreciated to know what upgrades a player has.

### Other Methods
The main idea here is to make it impossible to know if the cooldown on an upgrade was skipped. Having a delay like I propose will do that, but it's also an option to just update player's unlocked upgrades once an IRL month or two, while also removing timestamps.

**Comment by lunaynx** [comment]
I agree, this would be nice. I don't think it's *that* important to go to lengths to delay the visibility to make it impossible for anyone to tell if an upgrade was fast tracked, just not explicitly mentioning it should be enough. But if they really want to, that's still better than nothing.

In fact, strangely enough, despite Hypixel's general policy seeming to be not exposing monetization info in the API, for profile upgrades, it currently already exposes exact timestamps for when upgrades were started and claimed, and a boolean indicating whether they were fast tracked or not.
